### PR TITLE
Cleanup QuestionDataSource and fix breadcrumb links issues

### LINKED ISSUE #12616: Breadcrumb links not working correctly in Nested Questions
**Describe the bug**
Breadcrumb links under question title of Nested Questions are not working correctly.

**To Reproduce**
1. Create a question on a schema-based database (can be Simple or Native) - save as "Q1"
2. Simple question > Saved Questions > "Q1" - save as "Q2"
3. It should look like this now:
image
   a. The source question link is to `/question#...hash...` instead of `/question/:id`, which means it's not possible to click and edit the source, it will always create a new question - have to use search to find the source question to edit it (this is also the case for non-schema databases)
   b. Notice the schema name is the same as the collection - it should be `schema_0` (this example)
   c. The schema link is also using the collection name, which leads to `404` error page

**Expected behavior**
Link to the source question ID, so it can be edited.
Not sure if schema is known for questions based on Native, but then it shouldn't be shown.

**Information about your Metabase Installation:**
Metabase 0.33.0 thru 0.35.4

**Severity**
Giving P2, since it can be difficult to find the exact linked source question, when there are hundreds and some with same/similar names.

### LINKED ISSUE #17961: Breadcrumb path with multiple joined tables only has a single link to the first table
**Is your feature request related to a problem? Please describe.**
When joining 2 or more tables (or even another existing query), there is text on top of the page (see screenshot) that shows the  different data involved in the join, however the link only redirect to the first table of the list.

**Describe the solution you'd like**
It would be more convenient to have one different link for each underlying table/query.
Current link: [Orders + Products + People](https://www.metabase.com/question#Orders)
Expected link: [Orders](https://www.metabase.com/question#Orders) + [Products](https://www.metabase.com/question#Products) + [People](https://www.metabase.com/question#People)

**How important is this feature to you?**
Quite low but I guess the workload for this request is low too so it would be convenient to have.

**Additional context**
image

### PR DESCRIPTION
**Pull request by kulyk**
This PR refactors the `QuestionDataSource` component, fixes #12616, fixes #17961, and also increases the Simple Mode font size (was supposed to be shipped with #18903, but I broke it at some point). `QuestionDataSource` is used in the query builder header and displays breadcrumb links to question data sources (database, schema, table, join tables, source question, etc.)

### To Verify

**Breadcrumb path with multiple joined tables only has a single link to the first table #17961**

1. Ask a question > Custom Question > Sample Dataset > Orders
2. Join Products and People tables
3. Save the question, exit the notebook editor
4. You should see the "Orders + People + Products" label in the header breadcrumbs
5. Click every table name, it should open the clicked table

**Breadcrumb links not working correctly in Nested Questions #12616**

1. Create a question on a schema-based database (can be Simple or Native) - save as "Q1"
2. Simple question > Saved Questions > "Q1" - save as "Q2"
3. You should not see Q1's collection name in data source breadcrumbs, it should be "DB / Q1"
4. Click "Q1", you should be navigated to `/question/:id-q1` instead of an ad-hoc query based on Q1

### Demo

<details>
<summary>Breadcrumb path with multiple joined tables only has a single link to the first table</summary>

**Before**

https://user-images.githubusercontent.com/17258145/141514930-cfe9a14c-d2f1-4511-83cb-3221e330b68a.mov

**After**

https://user-images.githubusercontent.com/17258145/141514946-574ec03b-78db-4553-bc00-84aac74de7b5.mov

</details>

<details>
<summary>Breadcrumb links not working correctly in Nested Questions</summary>

**Before**

https://user-images.githubusercontent.com/17258145/141514988-17d409d8-d138-4615-90e8-7c10bfbb92ac.mov

**After**

https://user-images.githubusercontent.com/17258145/141514993-34633337-1981-46ce-9c66-b251221b3327.mov

</details>

### CODE REVIEW

**Comment by kulyk** [comment]
@metabase-bot run visual tests

**Comment by kulyk** [comment]
@metabase-bot run visual tests

**Comment by kulyk** [comment]
There was a color issue I missed while recording demos. Notice how table names in the header have different colors. All of them should actually be `text-dark`. This is fixed in the latest commit

CleanShot 2021-11-15 at 11 54 45@2x

**Review by alxnddr — APPROVED** [review]

File: frontend/src/metabase/lib/saved-questions/saved-questions.js [hunk 1]
```diff
@@ -29,6 +29,18 @@ export function getQuestionVirtualTableId(card) {
   return `card__${card.id}`;
 }
 
+export function isVirtualCardId(tableId) {
```
[the remark this answers is not in the document]
Comment (kulyk) [reply frontend/src/metabase/lib/saved-questions/saved-questions.js:32]: Good call, will do 👍

**Review by flamber — APPROVED** [review]
How wonderful useful the breadcrumbs have become!

**Pull request merged**